tests/test_cms_menus.py

```diff
@@ -391,6 +392,158 @@ def test_published_menu_in_preview_mode(self):
+
+    def test_published_page_in_menu_links_in_live_mode(self):
+        """
+        The live mode should show live links for any menu items attached to a page
+        """
+        menu_item_pagecontent = factories.PageContentWithVersionFactory(
+            language=self.language,
+            version__created_by=self.get_superuser(),
+            version__state=PUBLISHED,
+        )
+        factories.ChildMenuItemFactory(parent=self.published_menu_content.root, content=menu_item_pagecontent.page)
+
+        with self.login_user_context(self.get_superuser()):
+            response = self.client.get(self.live_endpoint)
+
+        expected_live_endpoint = menu_item_pagecontent.get_absolute_url(language=self.language)
+        not_expected_edit_endpoint = get_object_edit_url(menu_item_pagecontent, language=self.language)
+        not_expected_preview_endpoint = get_object_preview_url(menu_item_pagecontent, language=self.language)
+        actual_link = self._get_first_navigation_node_link_from_response(response)
+
+        self.assertEqual(response.status_code, 200)
+        self.assertIn(expected_live_endpoint, actual_link)
+        self.assertNotIn(not_expected_edit_endpoint, actual_link)
+        self.assertNotIn(not_expected_preview_endpoint, actual_link)
+
+    def test_unpublished_page_in_menu_links_in_preview_mode(self):
+        """
+        The preview should show no link for any menu items attached to an unpublished page
+        """
+        menu_item_pagecontent = factories.PageContentWithVersionFactory(
+            language=self.language,
+            version__created_by=self.get_superuser(),
+            version__state=UNPUBLISHED,
+        )
+        factories.ChildMenuItemFactory(parent=self.published_menu_content.root, content=menu_item_pagecontent.page)
+
+        with self.login_user_context(self.get_superuser()):
+            response = self.client.get(self.preview_endpoint)
+
+        not_expected_edit_endpoint = get_object_edit_url(menu_item_pagecontent, language=self.language)
+        not_expected_preview_endpoint = get_object_preview_url(menu_item_pagecontent, language=self.language)
+        not_expected_live_endpoint = menu_item_pagecontent.get_absolute_url(language=self.language)
+        actual_link = self._get_first_navigation_node_link_from_response(response)
+
+        self.assertEqual(response.status_code, 200)
+        self.assertNotIn(not_expected_edit_endpoint, actual_link)
+        self.assertNotIn(not_expected_preview_endpoint, actual_link)
+        self.assertNotIn(not_expected_live_endpoint, actual_link)
+
+    def test_archived_page_in_menu_links_in_preview_mode(self):
+        """
+        The preview should show no link for any menu items attached to an archived page
+        """
+        menu_item_pagecontent = factories.PageContentWithVersionFactory(
+            language=self.language,
+            version__created_by=self.get_superuser(),
+            version__state=UNPUBLISHED,
```

Comment:
Should this be `ARCHIVED`?

Reply:
As this is a factory it's already a method of sorts. We are only passing in what we want to control.

Reply:
It should have been ARCHIVED, well spotted. GitHuh has a really cool suggestions tool for small suggestions like this, I'll even use it myself for simplicity sometimes.

```suggestion
            version__state=ARCHIVED,
```